### PR TITLE
Gridss purple linx

### PR DESCRIPTION
**Pull request by lkhilton**
This PR contains updated rules to separate preprocessing of unmatched normals from tumour or tumour/normal pairs. These changes will go on version 1.0 since they only change the workflow logic and none of the outputs.

### CODE REVIEW

**Comment by rdmorin** [comment]
I will gladly review this

File: modules/gridss/1.0/gridss.smk [hunk 1]
```diff
@@ -285,26 +376,27 @@ rule _gridss_viral_annotation:
         viral_ref = str(rules._gridss_input_viral_ref.output), 
         viral_img = str(rules._gridss_setup_viral_ref.output)
     output: 
-        vcf = CFG["dirs"]["viral_annotation"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/gridss_viral_annotation.vcf.gz" 
+        vcf = temp(CFG["dirs"]["viral_annotation"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/gridss_viral_annotation.vcf.gz")
```
Comment (rdmorin) [inline modules/gridss/1.0/gridss.smk:338]: Won't this delete the main output?

File: modules/gridss/1.0/gridss.smk [hunk 2]
```diff
@@ -359,14 +453,15 @@ rule _gridss_run_gripss:
...
+        mem_mb = lambda wildcards, resources: int(resources.mem_mb * 0.8)
     conda: 
         CFG["conda_envs"]["gripss"]
     threads: 
         CFG["threads"]["gripss"]
     shell: 
+        # 'touch {output} && sleep 15'
         op.as_one_line(""" 
-        gripss -Xms4G -Xmx{params.jvmheap}m 
+        gripss -Xms4G -Xmx{params.mem_mb}m
```
Comment (rdmorin) [inline modules/gridss/1.0/gridss.smk:419]: What does -Xms4G do?

File: modules/gridss/1.0/gridss.smk [hunk 3]
```diff
@@ -388,6 +483,7 @@ rule _gridss_filter_gripss:
     conda: 
         CFG["conda_envs"]["bcftools"]
     shell: 
+        # 'touch {output} && sleep 15'
```
Comment (rdmorin) [inline modules/gridss/1.0/gridss.smk]: Why is this here?

File: modules/gridss/1.0/gridss.smk [hunk 4]
```diff
@@ -403,6 +499,7 @@ rule _gridss_gripss_to_bedpe:
     conda: 
         CFG["conda_envs"]["svtools"]
     shell: 
+        # 'touch {output} && sleep 15'
```
Comment (rdmorin) [inline modules/gridss/1.0/gridss.smk]: remove

File: modules/gridss/1.0/gridss.smk [hunk 5]
```diff
@@ -141,50 +146,112 @@ rule _gridss_input_bam:
...
-rule _gridss_symlink_preprocess: 
+rule _gridss_symlink_preprocessed_normal: 
     input: 
-        workdir = str(rules._gridss_preprocess.output.workdir)
+        workdir = str(rules._gridss_preprocess_unmatched_normal.output.workdir)
     output: 
-        workdir = directory(CFG["dirs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{sample_id}.bam.gridss.working")
-    group: "enormous_bam"
+        workdir = temp(CFG["dirs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{sample_id}.bam.gridss.working")
+        # workdir = CFG["dirs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{sample_id}.bam.gridss.working"
```
Comment (rdmorin) [inline modules/gridss/1.0/gridss.smk]: remove

File: modules/gridss/1.0/gridss.smk [hunk 6]
```diff
@@ -193,30 +260,41 @@ rule _gridss_paired:
...
-        n_workdir = CFG["dirs"]["preprocess"] + "{seq_type}--{genome_build}/{normal_id}.bam.gridss.working",
         fasta = str(rules._gridss_input_references.output.genome_fa),
         fasta_img = str(rules._gridss_setup_references.output.genome_img), 
         blacklist = reference_files("genomes/{genome_build}/encode/encode-blacklist.{genome_build}.bed"), 
         repeatmasker = reference_files("genomes/{genome_build}/repeatmasker/repeatmasker.{genome_build}.bed")
     output:
-        vcf = CFG["dirs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/gridss_raw.vcf.gz",
-        assembly = CFG["dirs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/assembly.bam"
+        vcf = temp(CFG["dirs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/gridss_raw.vcf.gz"),
+        assembly = CFG["dirs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/assembly.bam",
```
[the remark this answers is not in the document]
Comment (rdmorin) [reply modules/gridss/1.0/gridss.smk:248]: I think I understand. It's because the assembly directory is the working (not final) directory of the assembly

File: modules/gridss/1.0/gridss.smk [hunk 7]
```diff
@@ -193,30 +260,41 @@ rule _gridss_paired:
...
         steps = "assemble,call", 
-        jvmheap = lambda wildcards, resources: int(resources.mem_mb * 0.8) 
+        mem_mb = lambda wildcards, resources: int(resources.mem_mb * 0.8) 
     conda:
         CFG["conda_envs"]["gridss"]
     threads:
         CFG["threads"]["gridss"]
     resources:
         **CFG["resources"]["gridss"]
-    priority: 2
```
Comment (rdmorin) [inline modules/gridss/1.0/gridss.smk:216]: remove

**Review by rdmorin — APPROVED** [review]

File: modules/gridss/1.0/gridss.smk [hunk 8]
```diff
@@ -141,50 +146,112 @@ rule _gridss_input_bam:
...
-        jvmheap = lambda wildcards, resources: int(resources.mem_mb * 0.8) 
+        mem_mb = lambda wildcards, resources: int(resources.mem_mb * 0.8) 
     conda:
         CFG["conda_envs"]["gridss"]
     threads:
         CFG["threads"]["gridss"]
     resources:
         **CFG["resources"]["gridss"]
-    group: "enormous_bam"
+    priority: 1
```
[the remark this answers is not in the document]
Comment (lkhilton) [reply modules/gridss/1.0/gridss.smk:168]: I remember now that this actually encourages unmatched normal preprocessing to occur first, so I will keep it.

File: modules/gridss/1.0/gridss.smk (hunk 1 above)
Comment (lkhilton) [reply modules/gridss/1.0/gridss.smk:338]: Fortunately not. In unpaired mode the next step is to filter for only PASS variants and that's the only VCF I want to keep (they can be quite large). For paired mode gripss runs next.

File: modules/gridss/1.0/gridss.smk (hunk 2 above)
Comment (lkhilton) [reply modules/gridss/1.0/gridss.smk:419]: It's recommended by the devs, apparently it sets an initial java memory allocation? https://stackoverflow.com/questions/14763079/what-are-the-xms-and-xmx-parameters-when-starting-jvm

File: modules/gridss/1.0/gridss.smk (hunk 3 above)
Comment (lkhilton) [reply modules/gridss/1.0/gridss.smk]: This was group job debugging, will delete throughout.

**Comment by lkhilton** [comment]
@rdmorin I'm finished updating per your comments and this is ready for final review/approval.

**Review by rdmorin — APPROVED** [review]
